### PR TITLE
Properly handle `null` field values when encoding requests.

### PR DESCRIPTION
**Pull request by KOGI**
This fixes Issue #75

### CODE REVIEW

File: lib/src/utils.dart [hunk 1]
```diff
@@ -30,6 +30,10 @@ Map/*<K, V>*/ updateMap/*<K, V>*/(
 ///     mapToQuery({"foo": "bar", "baz": "bang"});
 ///     //=> "foo=bar&baz=bang"
 String mapToQuery(Map<String, String> map, {Encoding encoding}) {
+  map.keys
```
Comment (kevmoo) [inline lib/src/utils.dart]: Create a local variable and do this in two steps. It'll be easier to read.

Also: please add a test that exercises this case.

**Comment by KOGI** [comment]
I signed it!

File: lib/src/utils.dart (hunk 1 above)
Comment (KOGI) [inline lib/src/utils.dart]: Both done. The tests were super simple changes, so I don't expect any problems, but, I don't have sdk v1.23 installed due to local requirements for 1.22, so I couldn't *actually* run the test suite.

**Comment by donny-dont** [comment]
@kevmoo I don't see any references to `mapToQuery` in `master`. Shouldn't this target the 0.11 branch?

**Comment by KOGI** [comment]
@donny-dont you're totally right, my bad. I just pushed a rebase and changed the PR base.

Also, not sure what happened to my cla... it *was* there...

**Comment by nex3** [comment]
See https://github.com/dart-lang/http/issues/75#issuecomment-301630690

**Pull request closed without merging**